core/actions/table.ts

```diff
@@ -553,7 +562,19 @@ export class Table extends ActionBuilder<dataform.Table> {
   public dependencies(value: Resolvable | Resolvable[]) {
     const newDependencies = Array.isArray(value) ? value : [value];
     newDependencies.forEach(resolvable => {
-      this.proto.dependencyTargets.push(resolvableAsTarget(resolvable));
+      let dependencyTarget = resolvableAsTarget(resolvable);
+      dependencyTarget.includeDependentAssertions = dependencyTarget.includeDependentAssertions===undefined ? this.dependOnDependencyAssertions : dependencyTarget.includeDependentAssertions;
+      const existingDependencies = this.proto.dependencyTargets.filter(dependency => isSameTarget(dependencyTarget, dependency) && dependency.includeDependentAssertions !== dependencyTarget.includeDependentAssertions)
```

Comment:
It's not particularly obvious to me why this is needed, could you explain?

IIUC:

A => B, where B has `includeDependentAssertions: true`, and A which generates inline assertions, has `includeDependentAssertions: false`.

Why does A need to have `includeDependentAssertions` to be true for this to be valid?

Reply:
This checks the case where one dependency A is added multiple times(in config.dependencies and in ref()). but with different value set to `includeDependentAssertions` flag. In this case we should throw an error. 

Consider following .sqlx file.
`config {
            type: "table",
            dependencies: [{name: "A", includeDependentAssertions: false}]
        }
select * from \${ref({name: "A", includeDependentAssertions: true})}
select 1 as test`